sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -168,6 +178,100 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
   }
 
   /**
+   * Find the database with the specified name. This throws an [[AnalysisException]] when no
+   * [[Database]] can be found.
+   */
+  override def findDatabase(dbName: String): Database = {
+    if (sessionCatalog.databaseExists(dbName)) {
+      makeDatabase(sessionCatalog.getDatabaseMetadata(dbName))
+    } else {
+      throw new AnalysisException(s"The specified database $dbName does not exist.")
```

Comment:
How about throwing `NoSuchDatabaseException`? We also have the other NoSuchXYZ exceptions. Then, when writing the test cases, we can check these specific exception types instead of the general `AnalysisException`